lib/src/components/NxTransferListHalf/__tests__/NxTransferListHalf.test.tsx

```diff
@@ -4,495 +4,537 @@
+            .queryAllByRole('button')[1]).toHaveTextContent('Remove All');
+        expect(quickRender({ showMoveAll: true, isSelected: null })
+            .queryAllByRole('button')[1]).toHaveTextContent('Remove All');
+        expect(quickRender({ showMoveAll: true, isSelected: false })
+            .queryAllByRole('button')[1]).toHaveTextContent('Transfer All');
+      });
 
-    expect(item).toExist();
-    expect(item).toMatchSelector('div');
+  it('fires onMoveAll when the "Remove All"/ "Transfer All" button is clicked', async function() {
+    const user = userEvent.setup(),
+        onMoveAll = jest.fn(),
+        moveBtn = quickRender({ onMoveAll, showMoveAll: true }).getByRole('button', { name: /remove all/i });
 
-    expect(select).toExist();
-    expect(select).toMatchSelector('label');
-    expect(select).toHaveText('foo');
+    expect(onMoveAll).not.toHaveBeenCalled();
+    await user.click(moveBtn);
+    expect(onMoveAll).toHaveBeenCalled();
+  });
 
-    expect(icon).toExist();
-    expect(icon).toHaveProp('icon', faTimesCircle);
+  it('passes onMoveAll an array containing every id if no filterValue is set', async function() {
+    const dataItems = [{
+      id: '1',
+      displayName: 'foo'
+    }, {
+      id: '2',
+      displayName: 'Foo'
+    }, {
+      id: '3',
+      displayName: 'bar'
+    }, {
+      id: '4',
+      displayName: 'foo'
+    }, {
+      id: '5',
+      displayName: 'Foo'
+    }, {
+      id: '6',
+      displayName: 'bar'
+    }];
 
-    expect(checkbox).toExist();
-    expect(checkbox).toHaveProp('type', 'checkbox');
+    const user = userEvent.setup(),
+        onMoveAll = jest.fn(),
+        moveBtn = quickRender({ items: dataItems, onMoveAll, showMoveAll: true })
+            .getByRole('button', { name: /remove all/i });
 
-    const otherItem = list.find('.nx-transfer-list__item').at(1);
+    expect(onMoveAll).not.toHaveBeenCalled();
+    await user.click(moveBtn);
 
-    expect(otherItem).toHaveText('baz');
+    expect(onMoveAll).toHaveBeenCalledWith(['1', '2', '3', '4', '5', '6']);
   });
 
-  it('does not render move icon and checkbox input when onItemChange is not provided', function() {
-    const items = [{
-          id: 1,
-          displayName: 'foo'
-        }],
-        componentDisableItemMove = getMounted({ items }),
-        select = componentDisableItemMove
-            .find('.nx-transfer-list__item-list .nx-transfer-list__item .nx-transfer-list__select');
+  it('renders items into the .nx-transfer-list__item-list', function() {
+    const dataItems = [{
+      id: '1',
+      displayName: 'foo'
+    }, {
+      id: '2',
+      displayName: 'baz'
+    }];
 
-    expect(select.find(NxFontAwesomeIcon)).not.toExist();
-    expect(select.find('input.nx-transfer-list__checkbox')).not.toExist();
+    const { container } = quickRender({ items: dataItems }),
+        itemList = container.querySelector('.nx-transfer-list__item-list') as HTMLElement,
```

Comment:
At this time, there are no roles assigned to `.nx-transfer-list__item-list` or `.nx-transfer-list__item`, so need to use `querySelector`. RSC-998 should improve roles, which will mean updating the tests

Reply:
> At this time, there are no roles assigned to .nx-transfer-list__item-list or .nx-transfer-list__item, so need to use querySelector. RSC-998 should improve roles, which will mean updating the tests

Please add either a comment on that ticket referencing this, and/or a code comment here referencing that ticket, to try to prevent us from forgetting

Reply:
> Please add either a comment on that ticket referencing this, and/or a code comment here referencing that ticket, to try to prevent us from forgetting

Added on the ticket here: https://issues.sonatype.org/browse/RSC-988, and in the code here: [76ab1ca](https://github.com/sonatype/sonatype-react-shared-components/pull/880/commits/76ab1ca64f8cd9e8c7af3be8d0dce01bb7ee7706)